src/verso.rs

```diff
@@ -81,46 +82,55 @@ impl Verso {
     /// - Constellation: Enabled
     /// - Image Cache: Enabled
     pub fn new(evl: &ActiveEventLoop, proxy: EventLoopProxy<EventLoopProxyMessage>) -> Self {
-        let config = Config::new();
-        let to_controller_sender = if let Some(ipc_channel) = &config.args.ipc_channel {
-            let sender =
-                IpcSender::<ToControllerMessage>::connect(ipc_channel.to_string()).unwrap();
-            let (to_verso_sender, receiver) = ipc::channel::<ToVersoMessage>().unwrap();
-            sender
-                .send(ToControllerMessage::SetToVersoSender(to_verso_sender))
-                .unwrap();
-            let proxy_clone = proxy.clone();
-            ROUTER.add_typed_route(
-                receiver,
-                Box::new(move |message| match message {
-                    Ok(message) => {
-                        if let Err(e) =
-                            proxy_clone.send_event(EventLoopProxyMessage::IpcMessage(message))
-                        {
-                            log::error!("Failed to send controller message to Verso: {e}");
+        let cli_args = parse_cli_args().unwrap_or_default();
+        let (to_controller_sender, initial_settings) =
+            if let Some(ipc_channel) = &cli_args.ipc_channel {
+                let sender =
+                    IpcSender::<ToControllerMessage>::connect(ipc_channel.to_string()).unwrap();
+                let (to_verso_sender, receiver) = ipc::channel::<ToVersoMessage>().unwrap();
+                sender
+                    .send(ToControllerMessage::SetToVersoSender(to_verso_sender))
+                    .unwrap();
+                let ToVersoMessage::SetConfig(initial_settings) = receiver
+                    .recv()
+                    .expect("Failed to recieve the initial settings from controller")
+                else {
+                    panic!(
+                    "The initial message sent from versoview is not a `ToVersoMessage::SetConfig`"
+                )
+                };
+                let proxy_clone = proxy.clone();
+                ROUTER.add_typed_route(
+                    receiver,
+                    Box::new(move |message| match message {
+                        Ok(message) => {
+                            if let Err(e) = proxy_clone
+                                .send_event(EventLoopProxyMessage::IpcMessage(Box::new(message)))
+                            {
+                                log::error!("Failed to send controller message to Verso: {e}");
+                            }
```

Comment:
I feel like if you want to refactor how config workflow, this should also move to config module. And there could be an additional settings to get the controller sender.

Reply:
I don't think we can, it needs the event loop proxy, and except from `ToVersoMessage::SetConfig(initial_settings) = receiver`, all other code are not related to the config

Reply:
Or at least we can extract it into a function.
```rust
let (to_controller_sender, initial_settings) = ...
```

Another approach could be to give a dedicated channel to get the initial config first (`ToControllerMessage::GetInitConfig(IpcSender<ConfigFromController>)`) and then set ToVersoSender later so that we can decouple getting the config and setting the sender, but it seems a little bit inefficient.

Reply:
Made it to a function